.github/workflows/publish_release.yml

```diff
@@ -0,0 +1,89 @@
+name: Publish the latest release on PyPI
+
+on:
+  push:
+    tags:
+      - 'v*'
+    branches:
+      - main  # Restricts the workflow to only run on tags pushed to the main branch
+
+jobs:
+  validate-tag:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Validate Tag Format
+        run: |
+          TAG_REGEX='^v[0-9]+\.[0-9]+\.[0-9]+(-[a-zA-Z0-9\-\.]+)?$'
+          echo "Regex to match: $TAG_REGEX"
+          TAG="${GITHUB_REF/refs\/tags\//}"
+          echo "Tag extracted: $TAG"
+          if [[ "$TAG" =~ $TAG_REGEX ]]; then
+            echo "Tag format is valid."
+          else
+            echo "::error::Tag $TAG does not match the 'vMAJOR.MINOR.PATCH' or 'vMAJOR.MINOR.PATCH-pre-release' format."
+            exit 1
+          fi
+        shell: bash
+
+  build-and-package:
+    needs: validate-tag
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install poetry
+          poetry install
+
+      # Comment out the pytest step
+      # - name: Run tests
+      #   run: |
+      #     poetry run pytest
+
+      - name: Build the distribution
+        run: poetry build
+
+  publish-to-pypi:
+    needs: build-and-package
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
```

Comment:
looks like overlap and repeating previous steps, not sure this is what you have designed on the workflow

Reply:
It is just a duplication. I will remove this.

Reply:
its stil repeating, im not sure why this has to be separate from the first,